pg_metadata/install/sql/pgmetadata/10_FUNCTION.sql

```diff
@@ -110,6 +124,50 @@ BEGIN
         -- geometry_type
         NEW.geometry_type = test_geom_column.type;
 
+    ELSIF test_rast_column is not null THEN
+
+        -- column name
+        rast_column_name = test_rast_column.r_raster_column;
+        RAISE NOTICE 'pgmetadata - table % has a raster column: %', target_table, rast_column_name;
+
+        -- TODO: is this the best way to get the raster extent? Could we use ST_xmin etc. on test_rast_column.extent?
+        EXECUTE '
+            SELECT CONCAT(
+                min(ST_xmin(extent))::text, '', '',
+                max(ST_xmax(extent))::text, '', '',
+                min(ST_ymin(extent))::text, '', '',
+                max(ST_ymax(extent))::text)
+            FROM raster_columns
+            WHERE r_table_schema=' || quote_literal(NEW.schema_name) ||
+              'AND r_table_name=' || quote_literal(NEW.table_name)
+        INTO NEW.spatial_extent;
+
+        -- convexhull from target table
+        EXECUTE '
+            SELECT ST_Transform(ST_ConvexHull("' || rast_column_name || '"), 4326)
```

Comment:
Have you tested it with big rasters ? I fear this would be very slow for heavy data.

I think we could approximate this geometry by just using a rectangle built from the statistic extent taken from `raster_columns`.

Reply:
Thanks for having a look at this! So far I haven’t experienced performance issues. I didn’t try with really huge or highly irregular rasters, though.
`ST_ConvexHull()` doesn‘t treat NoData values as “outside the raster“, so it didn’t expect it to be too complicated.
We could use [`ST_Envelope()`](https://postgis.net/docs/RT_ST_Envelope.html), which returns just the X/Y Min/Max rectangle.